openedx/features/course_search/static/course_search/js/collections/search_collection.js

```diff
@@ -40,9 +40,18 @@
                     },
                     type: 'POST',
                     success: function(self) {
+                        self.errorMessage = '';  // eslint-disable-line no-param-reassign
                         self.trigger('search');
                     },
-                    error: function(self) {
+                    error: function(self, response) {
+                        if (response.responseJSON && response.responseJSON.error) {
```

Comment:
@ciuin This code appears multiple times - does in make sense to put it into common ancestor (`search_state`?) or helper class?

Note in `search_form.js` it's essentially the same if you pass `response.responseJSON` there - so it could also be considered duplicate.